sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUDF.scala

```diff
@@ -1057,5 +1057,17 @@ case class ScalaUDF(
 
   private[this] val converter = CatalystTypeConverters.createToCatalystConverter(dataType)
 
-  override def eval(input: InternalRow): Any = converter(f(input))
+  val npeErrorMessage = "Given UDF throws NPE during execution, please check the UDF " +
+    "to make sure it handles null parameters correctly."
+
+  override def eval(input: InternalRow): Any = {
+    val result = try {
+      f(input)
+    } catch {
+      case e: NullPointerException =>
+        throw new RuntimeException(npeErrorMessage, e)
```

Comment:
Can we still use `NullPointerException`? NullPointerException can have a specified message. Then, you can use `setStackTrace` to set the original stacktrace.